doc/running-cljdoc-locally.adoc

```diff
@@ -0,0 +1,461 @@
+Fork it if you want to walk though pushing changes for a full preview.
+
+==== Step 1) Commit/Push Library Changes
+Cljdoc only imports from committed changes, this means you'll need to commit any changes before import.
+
+If you want a full preview where direct GitHub references (links to images, source code, etc) work, then you'll also want to push your commits to GitHub.
+
+==== Step 2) Publish Library Locally
+After <<setup>>, to publish {example-project-name} to your local maven repository, run:
+
+[source,shell,subs="verbatim,attributes"]
+----
+{example-project-local-install}
+----
+
+The command issued to install a library to the local maven repo varies by build technology (leiningen, boot, tools deps cli, etc) and project.
+The `{example-project-local-install}` command is appropriate for {example-project-name}, be sure to use the appropriate command for your project.
+
+==== Step 3) Import Library Docs
+We go over a few different import scenarios:
+
+[[import-a-published-library,import a published library]]
+===== Import a Published Library
+You want to import library docs for a publicly published library.
+
+This is a likely scenario for a *cljdoc developer*.
+
+TIP: If you have been testing {example-project-name} locally, be sure to delete it from your local maven repository.
+
+.cmd for cljdoc source
+[%collapsible]
+====
+=====
+[source,shell,subs="verbatim,attributes"]
+----
+../cljdoc/script/cljdoc ingest \
+  --project {example-project-coords} \
+  --version {example-project-version}
+----
+=====
+====
+
+.cmd for cljdoc docker
+[%collapsible]
+====
+=====
+[source,shell,subs="verbatim,attributes"]
+----
+docker run --rm \
+  --volume "$HOME/.m2:/root/.m2" \
+  --volume /tmp/cljdoc:/app/data \
+  --entrypoint clojure \
+  {docker-image} -A:cli ingest \
+    --project {example-project-coords} \
+    --version {example-project-version}
+----
+=====
+====
+
+
+===== Testing docs for a Library During Development
+You are in development mode and want to preview what your library docs will look like on cljdoc.
+
+Normally cljdoc will look in your published library's `pom.xml` under `project/scm` tag to learn what GitHub `url` and `tag` it should use to reference articles and source code.
+It can be convenient to not have to alter your `pom.xml` while testing changes during development.
+
+If this is true for you, the cljdoc `ingest` command allows the `pom.xml`:
+
+* `project/scm/url` to be overridden via `--git`
+* `project/scm/tag` to be overridden via `--rev`
+
+For `--git` you can specify either:
+
+* Your library's root dir on your local filesystem for a quick preview. +
+In a quick preview, direct references to GitHub won't work.
+* The https GitHub project URL for a full preview. +
+In a full preview, direct references to GitHub will work.
+
+From {example-project-name}'s root dir:
+
+.cmd for cljdoc source - quick preview
+[%collapsible]
```

Comment:
These code blocks are the core of this guide and in my personal experience I often would just look at the snippets, see if they make sense and run them without actually reading the rest. With that in mind I have two suggestions:

- Don't hide them by default
- Use a table to contrast Docker/Local approach 🐳  The code snippets seem narrow enough that a table could work (at least on non-mobile screens).

Reply:
This one was tricky for me and I went through more variations than I'd like to admit. 🙂 

I felt the hidden code blocks allowed the headings and descriptive text (and therefore outline) to be visible on the page.  This variation felt less overwhelming to me and I got less lost on the page.  

But these are subjective things. 

Let me know if you feel strongly about this and I can experiment more.

Reply:
I do think I see your point now. I think I was just getting tired of editing this doc.  I'll give it a shot.

Reply:
Thanks for persevering 🙇